File: abc/abc157/abc157_c.cs
// https://atcoder.jp/contests/abc157/submissions/10455997

using System;
using System.Linq;

namespace CSharp
{
    class Program
    {
        static void Main()
        {
            var input = Console.ReadLine().Split().Select(int.Parse).ToArray();
            var N = input[0];
            var M = input[1];

            var s = new int[M][];
            for (int i = 0; i < M; i++)
            {
                s[i] = Console.ReadLine().Split().Select(int.Parse).ToArray();
            }

            for (int i = 0; i < M; i++)
            {
                if (N > 1 && s[i][0] == 1 && s[i][1] == 0)
                {
                    Console.WriteLine("-1");
                    return;
                }
                for (int j = 0; j < M; j++)
                {
                    if (s[i][0] == s[j][0] && s[i][1] != s[j][1])
                    {
                        Console.WriteLine("-1");
                        return;
                    }
                }
            }

            var c = new char[N];
            for (int i = 0; i < N; i++)
            {
                c[i] = '0';
            }
            for (int i = 0; i < M; i++)
            {
                c[s[i][0] - 1] = char.Parse(s[i][1].ToString());
            }
            if (N > 1 && c[0] == '0')
            {
                c[0] = '1';
            }
            Console.WriteLine(int.Parse(string.Join(string.Empty, c)));
        }
    }
}

File: abc/abc157/abc157_c.fs
// https://atcoder.jp/contests/abc157/submissions/11871973

open System.Linq
open System.Collections.Generic

let [| N; M |] = stdin.ReadLine().Split() |> Array.map int

let S = new List<int []>()

for i in [ 1..M ] do
    stdin.ReadLine().Split()
    |> Array.map int
    |> S.Add

let result = new List<string>()

for i in [ 0..999 ] do
    let s = i.ToString()
    if s.Length = N then
        if S.All(fun x -> (s.[x.[0] - 1] |> int) - ('0' |> int) = x.[1]) then s |> result.Add

if result.Count > 0 then result.[0]
else "-1"
|> stdout.WriteLine
